src/python_testing/TC_DISHALM_3_2.py

```diff
@@ -0,0 +1,139 @@
+
+        logging.info("Notify: %s" % (notify))
+        logging.info("Notify State: %s" % (notify_state))
+
+        asserts.assert_true(notify_state & Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kDrainError, "State bit 1 is not set to TRUE")
+
+        self.print_step(4, "2b. Read from the DUT the State Attribute")
+        state = self.read_state_attribute(endpoint=endpoint)
+
+        logging.info("State: %s" % (state))
+
+        asserts.assert_true(state & Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kDrainError, "Bit 1 of State is not set to 1")
+
+        self.print_step(5, "2c. Operate Device to set the condition to lower the Drain alarm and wait a few seconds")
+        input("Press Enter when done.\n")
+
+        notify = self.read_event_notify(endpoint=endpoint)
+        notify_state = self.read_event_notify_state(endpoint=endpoint)
+
+        logging.info("Notify: %s" % (notify))
+        logging.info("Notify State: %s" % (notify_state))
+
+        latch = self.read_latch_attribute(endpoint=endpoint)
+        logging.info("Latch: %s" % (latch))
+
+        if latch == 0:
+            asserts.assert_false(notify_state & Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kDrainError,
+                                 "State bit 1 is not set to FALSE")
+
+        self.print_step(6, "2d. Read from the DUT the State Attribute")
+        state = self.read_state_attribute(endpoint=endpoint)
+
+        logging.info("State: %s" % (state))
+
+        latch = self.read_latch_attribute(endpoint=endpoint)
+        logging.info("Latch: %s" % (latch))
+
+        if latch == 0:
+            asserts.assert_false(state & Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kDrainError, "Bit 1 of State is not set to 0")
+        elif latch == 1:
+            asserts.assert_true(state & Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kDrainError, "Bit 1 of State is not set to 1")
+
+        self.print_step(7, "3a. Send to the DUT the Reset Command with bit 1 of Alarms set to 1")
+        alarm = Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kDrainError
+
+        await self.send_reset_cmd(setAlarm=alarm)
+
+        notify = self.read_event_notify(endpoint=endpoint)
+        notify_state = self.read_event_notify_state(endpoint=endpoint)
+
+        logging.info("Notify: %s" % (notify))
+        logging.info("Notify State: %s" % (notify_state))
+
+        latch = self.read_latch_attribute(endpoint=endpoint)
+        logging.info("Latch: %s" % (latch))
+
+        if latch == 1:
+            asserts.assert_false(notify_state & Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kDrainError,
+                                 "State bit 1 is not set to FALSE")
+
+        self.print_step(8, "3b. Read from the DUT the State Attribute")
+        state = self.read_state_attribute(endpoint=endpoint)
+
+        logging.info("State: %s" % (state))
+
+        asserts.assert_false(state & Clusters.DishwasherAlarm.Bitmaps.AlarmMap.kDrainError, "Bit 1 of State is not set to 0")
+
```

Comment:
Again, `4a` onwards not here - but starting to think its maybe because we can't test that?